src/org/jitsi/impl/neomedia/transform/rewriting/Utils.java

```diff
@@ -0,0 +1,221 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jitsi.impl.neomedia.transform.rewriting;
+
+import org.jitsi.impl.neomedia.codec.*;
+import org.jitsi.impl.neomedia.*;
+import org.jitsi.service.neomedia.codec.Constants;
+import org.jitsi.util.*;
+
+/**
+ * This class contains utility methods for video codecs.
+ *
+ * @author George Politis
+ */
+public class Utils
+{
+  /**
+   * The {@link Logger} used by the {@link Utils} class for logging output.
+   */
+  private static final Logger logger = Logger.getLogger(Utils.class);
+
+
+  private static org.jitsi.impl.neomedia.codec.video.h264.DePacketizer h264DePacketizer =
+        new org.jitsi.impl.neomedia.codec.video.h264.DePacketizer();
+
+  private static org.jitsi.impl.neomedia.codec.video.vp8.DePacketizer vp8DePacketizer =
+        new org.jitsi.impl.neomedia.codec.video.vp8.DePacketizer();
+
+  /**
+   * Utility method that determines whether or not a packet is a key frame.
+   *
+   * @param pkt The <tt>RawPacket</tt> to determine whether it is a key frame
+   * or not.
+   * @param codecPT The codec payload type
+   * @param redPT The RED payload type.
+   * @return true if the packet is a VP8 key frame, false otherwise.
+   */
+  public static boolean isKeyFrame(RawPacket pkt, Byte redPT, Byte codecPT, String targetCodec)
+  {
+      if (pkt == null)
+      {
+          return false;
+      }
+      // We are calling keyframe detection for all codec types here and one
+      // should only work
+      if (targetCodec.equalsIgnoreCase(Constants.VP8))
+      {
+          if(isVP8KeyFrame(
+              pkt.getBuffer(), pkt.getOffset(), pkt.getLength(), redPT, codecPT))
+          {
+              return true;
+          }
+      }
+      else if (targetCodec.equalsIgnoreCase(Constants.H264))
+      {
+          if (isH264KeyFrame(
+              pkt.getBuffer(), pkt.getOffset(), pkt.getLength(), redPT, codecPT))
+          {
+              return true;
+          }
+      }
+      else
+      {
+          logger.error("Unsupported Codec type: " + codecPT);
+      }
+      return false;
+  }
+
+  /**
+   * Utility method that determines whether or not a packet is a key frame.
+   *
+   * @param buf the buffer that holds the RTP payload.
+   * @param off the offset in the buff where the RTP payload is found.
+   * @param len then length of the RTP payload in the buffer.
+   * @param redPT The RED payload type.
+   * @param vp8PT The VP8 payload type.
+   * @return true if the packet is a VP8 key frame, false otherwise.
+   * @return true if the packet is a VP8 key frame, false otherwise.
+   */
+  public static boolean isVP8KeyFrame(
+      byte[] buf, int off, int len, Byte redPT, Byte vp8PT)
```

Comment:
Since `isVP8KeyFrame` and `isH264KeyFrame` are almost identical, would it make sense to factor out the part that finds the buffer, offset, length of the payload and then call the appropriate depacketizer?